connection.go

```diff
@@ -268,9 +275,10 @@ func (conn *Connection) checkTimeout() error {
 				case conn.session.resendChan <- seq:
 					conn.resentSeq[seq] = struct{}{}
 					conn.windowSize /= 2
-					if conn.windowSize < uint32(conn.session.config.MinConnectionWindowSize) {
-						conn.windowSize = uint32(conn.session.config.MinConnectionWindowSize)
+					if conn.windowSize < float64(conn.session.config.MinConnectionWindowSize) {
```

Comment:
Use setWindowSize() to avoid repeated code

Reply:
Yes, modified